pkg/operator/ceph/file/controller.go

```diff
@@ -200,7 +200,7 @@ func (r *ReconcileCephFilesystem) reconcile(request reconcile.Request) (reconcil
 	}
 
 	// DELETE: the CR was deleted
-	if !cephFilesystem.GetDeletionTimestamp().IsZero() {
+	if !cephFilesystem.GetDeletionTimestamp().IsZero() && (len(cephFilesystem.Spec.DataPools) > 0 || len(cephFilesystem.Spec.DataPools) > 0 || cephFilesystem.Spec.MetadataServer.ActiveCount > 0) {
```

Comment:
Instead of adding these checks, we just want to ignore the error on line 205 if `reconcileDeleteFilesystem()` detects that the filesystem doesn't exist. For example, a couple layers into that function, there is a call to `client.GetFilesystem()`, which might not find the filesystem. If not found, we should be able to differentiate between that error and other errors, then return success here.